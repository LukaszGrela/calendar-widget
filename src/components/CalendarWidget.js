import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import CalendarRow from './CalendarRow';
import CalendarMonthGrid from './CalendarMonthGrid';

class CalendarWidget extends React.Component {

    render = () => {
        const now = moment();
        console.log(now)
        console.log(moment.weekdaysShort())
        return (
            <div className='CalendarWidget'>
                <div className='month-page'>
                    <div className='calendar-navigation'>

                    </div>
                    <div className='calendar-view'>
                        <CalendarRow
                            className='week-header'
                            days={moment.weekdaysShort()} />

                        <CalendarMonthGrid
                            now={now} />
                    </div>
                </div>
            </div>
        );
    }
}

CalendarWidget.propTypes = {

}

export default CalendarWidget;