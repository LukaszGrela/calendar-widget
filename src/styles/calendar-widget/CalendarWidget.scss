$primary-color:#f1f1f1;
$color-blue:rgb(107, 164, 221);
$color-dark-grey:#2f2f2f;
$squareRatio: percentage(1/7);
$squareRatioHalf: percentage(1/14);
.CalendarWidget {
    width: 80%;
    max-width: 64rem;
    margin: 3.2rem auto;
    .calendar-header {
        display: flex;
        justify-content: space-between;
        align-items: center;
        margin-left: 4%;
        margin-right: 4%;
        margin-bottom: 3.2rem;
        .today {
            color: $color-blue;
            font-size: 3.2rem;
        }
        .middle {
            order: 0;
        }
        .left {
            order: -1;
        }
        .right {
            order: 1;
        }
        .month-navigation {
            svg {
                width: 2.4rem;
                fill: $primary-color;
            }
            button {
                width: 3.2rem;
                height: 3.2rem;
                border: none;
                background: none;
            }
        }
        .year-month-selectors {
            select {
                outline: none;
                border: none;
                font-size: 1.6rem;
                background: $color-dark-grey;
                color: $primary-color;
            }
            select:hover {
                color: $color-blue
            }
        }
    }
    .row {
        display: flex;
    }
    .day {
        width: $squareRatio;
        padding-top: $squareRatioHalf - 2;
        padding-bottom: $squareRatioHalf - 2;
        text-align: center;
        font-weight: bold;
    }
    .calendar-view {
        display: flex;
        flex-direction: column;
    }
    .week-header .day {
        padding-top: 1%;
        padding-bottom: 1%;
        margin-bottom: 2%;
    }
    .month-grid .day {
        &:hover {
            outline: dashed 0.2rem $color-blue;
            outline-offset: -0.4rem;
        }
    }
    .month-grid .day.today {
        background-color: $color-blue;
        color: $color-dark-grey;
        outline: solid 0.2rem $color-dark-grey;
        outline-offset: -0.4rem;
        &:hover {
            outline-style: dashed;
        }
    }
    .spill {
        color: darken($color: $primary-color, $amount: 33);
        font-weight: normal;
    }
}