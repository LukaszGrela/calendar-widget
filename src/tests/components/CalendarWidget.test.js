import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import CalendarWidget from '../../components/CalendarWidget';


let wrapper, today, current;
beforeEach(() => {
    today = moment('19790613');
    current = moment();
    wrapper = shallow(<CalendarWidget
        todayDate={today}
        currentMonth={current}
    />);
})


test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call an action when clicked on today title', () => {

    wrapper.instance().today = jest.fn();
    wrapper.update();
    const todayTitle = wrapper.find('.today');
    expect(todayTitle).toHaveLength(1);
    todayTitle.at(0).simulate('click');
    expect(wrapper.instance().today).toBeCalled();
});