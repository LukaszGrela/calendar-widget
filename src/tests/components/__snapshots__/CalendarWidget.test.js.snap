// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="CalendarWidget undefined"
>
  <div
    className="month-page"
  >
    <div
      className="calendar-header"
    >
      <div
        className="today middle"
        onClick={[Function]}
      >
        June 13th 1979
      </div>
      <CalendarYearMonthSelectors
        className="left"
        month={0}
        monthChanged={[Function]}
        year={1970}
        yearChanged={[Function]}
      />
      <CalendarNavigation
        className="right"
        navigateDown={[Function]}
        navigateUp={[Function]}
      />
    </div>
    <div
      className="calendar-view"
    >
      <CalendarRow
        className="week-header"
        days={
          Array [
            "Sun",
            "Mon",
            "Tue",
            "Wed",
            "Thu",
            "Fri",
            "Sat",
          ]
        }
      />
      <CalendarMonthGrid
        date={"1970-01-01T00:00:00.000Z"}
        dayClicked={[Function]}
        now={"1979-06-12T23:00:00.000Z"}
      />
    </div>
  </div>
</div>
`;
